src/lib/packet.h

```diff
@@ -565,21 +572,44 @@ typedef enum {
 
 #define PGP_SA_DEFAULT_CIPHER PGP_SA_CAST5
 
+#define PGP_SA_DEFAULT_BLOCK_CIPHER_MODE PGP_BLOCK_CIPHER_MODE_CFB
+
 void pgp_calc_mdc_hash(
   const uint8_t *, const size_t, const uint8_t *, const unsigned, uint8_t *);
 unsigned pgp_is_hash_alg_supported(const pgp_hash_alg_t *);
 
+/* Maximum block size for symmetric crypto */
+#define PGP_MAX_BLOCK_SIZE 16
+
+/* Maximum key size for symmetric crypto */
+#define PGP_MAX_KEY_SIZE 32
+
+/* Salt size for hashing */
+#define PGP_SALT_SIZE 8
+
+#define PGP_NONCE_SIZE 12
+
+#define PGP_PROTECTED_AT_SIZE 15
+
+typedef struct pgp_key_t pgp_key_t;
+
+typedef struct pgp_seckey_t pgp_seckey_t;
+
+typedef pgp_seckey_t *pgp_seckey_decrypt_t(const pgp_key_t *key, FILE *passfp);
+
 /** pgp_seckey_t
  */
-typedef struct pgp_seckey_t {
+struct pgp_seckey_t {
     pgp_pubkey_t        pubkey; /* public key */
     pgp_s2k_usage_t     s2k_usage;
     pgp_s2k_specifier_t s2k_specifier;
-    pgp_symm_alg_t      alg;      /* symmetric alg */
-    pgp_hash_alg_t      hash_alg; /* hash algorithm */
+    pgp_symm_alg_t      alg;               /* symmetric alg */
+    pgp_block_cipher_t  block_cipher_mode; /* block cipher mode */
+    pgp_hash_alg_t      hash_alg;          /* hash algorithm */
     uint8_t             salt[PGP_SALT_SIZE];
     unsigned            s2k_iterations;
     uint8_t             iv[PGP_MAX_BLOCK_SIZE];
+    uint8_t             nonce[PGP_NONCE_SIZE];
```

Comment:
We don't really need a separate nonce field, just use the iv field. For OCB it will be 12 bytes, for CBC use the block size.